internal/ingress/controller/config/config.go

```diff
@@ -790,6 +795,7 @@ func NewDefault() Configuration {
 		ErrorLogLevel:                    errorLevel,
 		UseForwardedHeaders:              false,
 		EnableRealIP:                     false,
+		EnableRealIPRecursive:            true,
```

Comment:
shouldn't this be disabled by default?

Reply:
there is a "problem" with this approach of a flag defaulted to true: if you set it as false on your configmap, it may or may not represent as empty (depending on the marshaller, eg using omitempty on json flag) so because it is empty it will be defaulted to...true :) 

To confirm this behavior, can you add some e2e test with both EnableRealIPRecursive = true or false on the configmap, and check if the template is rendered correctly?

Reply:
Hi @rikatz ,

I've set the default to `false`. My initial reasoning for it to be `true` was that I think the more secure option should be enabled by default if it doesn't otherwise impact any functionality. Because the setting itself is dependent on `EnableRealIp` being set to `true`, the more secure configuration would be the default when `EnableRealIp` would be used. This is a taste thing, and not a deal breaker for getting this merged.